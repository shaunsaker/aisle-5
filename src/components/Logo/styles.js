import { StyleSheet, Platform } from 'react-native';

import styleConstants from '../../styleConstants';

const LETTER_SPACING = 5;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    ...styleConstants.fonts.types.title,
    color: styleConstants.colors.white,
    letterSpacing: LETTER_SPACING,
  },
  icon: {
    fontSize: 18,
    color: styleConstants.colors.secondary,
    marginLeft: Platform.OS == 'android' ? LETTER_SPACING : 0,
    marginRight: LETTER_SPACING,
  },
});

export default styles;
