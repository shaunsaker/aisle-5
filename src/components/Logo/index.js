import React from 'react';
import { View, Text } from 'react-native';

import styles from './styles';

import CustomIcon from '../../assets/icons';

const propTypes = {};

const defaultProps = {};

const Logo = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>A</Text>

      <CustomIcon name="logo" style={styles.icon} />

      <Text style={styles.text}>SLE 5</Text>
    </View>
  );
};

Logo.propTypes = propTypes;
Logo.defaultProps = defaultProps;

export default Logo;
