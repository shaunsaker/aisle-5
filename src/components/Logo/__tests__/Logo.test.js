import React from 'react';
import renderer from 'react-test-renderer';

import Logo from '..';

describe('Logo', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Logo />);

      expect(component).toMatchSnapshot();
    });
  });
});
