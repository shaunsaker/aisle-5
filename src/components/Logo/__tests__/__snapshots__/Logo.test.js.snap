// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#EDE8CA",
        "fontFamily": "ArchivoBlack-Regular",
        "fontSize": 24,
        "letterSpacing": 5,
        "lineHeight": 30,
      }
    }
  >
    A
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#FEA116",
          "fontSize": 18,
          "marginLeft": 0,
          "marginRight": 5,
        },
        Object {
          "fontFamily": "AppIcons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î¤€
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#EDE8CA",
        "fontFamily": "ArchivoBlack-Regular",
        "fontSize": 24,
        "letterSpacing": 5,
        "lineHeight": 30,
      }
    }
  >
    SLE 5
  </Text>
</View>
`;
